.github/scripts/promote_branch.sh

```diff
@@ -0,0 +1,160 @@
+#     script will not git push to add more content to the staging branch. This can be overridden with
+#     --force-to-staging true
+#   - If promoting to production and the content has not been in the staging branch for at least 7 days,
+#     the script will exit without doing a push. Content is expected to sit in staging for at least a week
+#     to provide sufficient testing time. This can be overridden with --override true
+#
+# Prerequisities:
+#   - An environment variable GITHUB_TOKEN is defined that provides access to the user's account. See
+#     https://github.com/redhat-appstudio/release-service-utils/blob/main/ci/promote-overlay/README.md#setup for help.
+#   - curl, git and jq installed.
+
+set -e
+
+# GitHub repository details
+ORG="redhat-appstudio"
+REPO="release-service-catalog"
+
+OPTIONS=$(getopt --long "target-branch:,force-to-staging:,override:,debug,help" -o "tgt:,h" -- "$@")
+eval set -- "$OPTIONS"
+while true; do
+    case "$1" in
+        -tgt|--target-branch)
+            TARGET_BRANCH="$2"
+            shift 2
+            ;;
+        --force-to-staging)
+            FORCE_TO_STAGING="$2"
+            shift
+            ;;
+        --override)
+            OVERRIDE="$2"
+            shift
+            ;;
+        --debug)
+            DEBUG="true"
+            shift
+            ;;
+        -h|--help)
+            print_help
+            exit
+            ;;
+        --)
+            shift
+            break
+            ;;
+        *) echo "Error: Unexpected option: $1" % >2
+    esac
+done
+
+print_help(){
+    echo "Usage: $0 --target-branch branch [--force-to-staging false] [--override false] [--debug]"
+    echo
+    echo "  --target-branch:    The name of the branch to be promoted. Options are staging"
+    echo "                      and production."
+    echo "  --force-to-staging: If passed with value true, allow promotion to staging even"
+    echo "                      if staging and production differ."
+    echo "  --override:         If passed with value true, allow promotion to production"
+    echo "                      even if the change has not been in staging for one week."
+    echo "  --debug:            For debugging purposes, do everything except for the git"
+    echo "                      push and deleting the temp repo."
+    echo
+    echo "  --target-branch has to be specified."
+}
+
+check_if_branch_differs() {
+    ACTUAL_DIFFERENT_LINES=$(git diff --numstat origin/$1 | wc -l)
+    if [ $ACTUAL_DIFFERENT_LINES -ne 0 ] ; then
+        echo "Lines differ in branch $1"
+        echo "Actual differing lines: $(git diff --numstat origin/$1)"
+        exit 1
+    fi
+}
+
+check_if_any_commits_in_last_week() {
+    NEW_COMMITS=$(git log --oneline --since="$(date --date="7 days ago" +%Y-%m-%d)" | wc -l)
+    if [ $NEW_COMMITS -ne 0 ] ; then
+        echo "There are commits in staging that are less than a week old. Blocking promotion to promotion"
```

Comment:
"Blocking promotion to promotion" -> to production

Reply:
Thanks, updated